os/os/__init__.py

```diff
@@ -142,18 +142,18 @@ def listdir(path="."):
 def walk(top, topdown=True):
     files = []
     dirs = []
-    for dirent in ilistdir_ex(top):
-        mode = dirent[3] << 12
-        fname = dirent[4].split(b'\0', 1)[0]
+    for dirent in ilistdir(top):
+        mode = dirent[1] << 12
+        fname = fsdecode(dirent[0])
         if stat_.S_ISDIR(mode):
-            if fname != b"." and fname != b"..":
```

Comment:
You changed semantics of this line, even though MicroPython's ContributorGuidelines explicitly warns against it. Your change does superfluous allocation which in turn leads to cases like https://github.com/micropython/micropython/issues/2057 .